sdploy/spack_yaml.py

```diff
@@ -145,6 +154,22 @@ def create_pkgs_specs_dict(self):
                 if len(self.pkgs_specs[pkg_list_name]['dependencies']) == 0:
                     self.pkgs_specs[pkg_list_name].pop('dependencies')
 
+    def _skip_list(self, pkg_list_name):
+        """Returns true if the package list passed in argument is to skip
+
+        This method searches for the presence of the 'filters' key in the
+        package configuration section. If found, it will read each filter
+        here defined and if at least one has the none property, this list
+        is skipped."""
+
+        _skip = False
+        pkg_list_cfg = self.pkgs_stack[pkg_list_name]
+        if 'filters' in pkg_list_cfg:
+            for filter in pkg_list_cfg['filters']:
+                if self.filters[filter] == 'none':
+                    _skip = True
+        return _skip
```

Comment:
```suggestion
                    return True
        return False
```

once you found a filter that turn skip to True you can do an early out. This suggestion also means you can remote the _skip variable